### DISCUSSION TITLE
Update Log for iTwin Synchronizer

### DISCUSSION THREAD

**Discussion by youngbrett** [iTwin/itwinjs-core#4225] [General] [answered]
Hi --

Is there a public update log for the iTwin Synchronizer and it's components?  I want to check whether changes made in a recent  iTwinJSCore release have been implemented in iTwin Synchronizer and related iModel Connectors.  [This page](https://www.bentley.com/en/resources/itwin-synchronizer) gives me a direct file link and I can't check versions without doing the download.

Thanks much,

Brett.

**Comment by pmconne** [comment]
@jffmarker @timlawrence-bentley?

**Reply by jffmarker** [reply]
I'm also curious why this information is important to you. The iTwin Synchronizer App itself does use imodel.js, but it updates and stabilizes on a version as the team deems necessary. Further, the Connectors it runs do not actually use imodel.js (or itwin.js). The version of dependencies like itwin.js is generally treated as an internal implementation detail today.

**Reply by youngbrett** [reply]
@jffmarker - I don't quite understand your explanation of how the iTwin Synchronizer fits into the greater architecture. Can you clarify how iTwin Synchronizer isn't a part of iTwin.js or iModel.js?  I do understand that there are good alternatives to the iTwin Synchronizer, like the Synchronization API, but have not moved there yet.  

My interest is I did some dev and changed a client's workflow on a project to get around a performance issue using the Presentation API with Autodesk Navisworks files (Service Request 7001427267).  Bentley said they would address the issue in a future iTwin.js release and, after talking with @mgooding last week, we thought it was fixed in iTwin.js 3.3.x. When I read the iTwin.js release notes, I got the feeling that the fix might require updates to the iTwin Synchronizer. It turned out to be a dead end because 3.3.x did not fix the issue but I think it is a good question.  Before today, I was using iTwin Synchronizer version 1.1.321 and I have no way to check the currency of that version except by doing a generically named download on a webpage with no version info. Because of a sync failure today using iTwin Synchronizer, I just did the download and I'm now on version 1.1.331.  Returning to my original question, I don't have any info on how my current version is different from my previous version - not even a readme in the installed directory.

(It says my service request for this bug got shifted to Github as an issue, but I couldn't find it.)

**Reply by jffmarker** [reply]
The version of iTwin.js that iTwin Synchronizer uses will not affect the conversion or display of data. Also, the Connectors that it orchestrates today do not use iTwin.js either. From the background you mention, the component that most likely matters is the Web viewer application (e.g. Design Review)... maybe @mgooding can confirm... so I'm guessing you'd want to follow that team's deployments and updates. If you are using Design Review, perhaps @jjbeckman13 can direct you to release notes for that App.

Perhaps another way of thinking about it... iTwin Synchronizer is a thin wrapper on top of the Connectors. Its job is merely to know how to call them; iTwin Synchronizer itself does not actually read, write, or view the iModel.

**Reply by mgooding** [reply]
In this case, the expected fix will be in the @itwin/core-backend package when it is eventually made. @khanaffan is still investigating a solution. The slow queries @youngbrett is seeing would be present in Design Review, but @youngbrett is seeing this and looking to resolve it in [imodel-unity-example](https://www.github.com/itwin/imodel-unity-example/).

Unless @khanaffan finds otherwise, the solution should not require updating Connectors or iTwin Synchronizer.

> (It says my service request for this bug got shifted to Github as an issue, but I couldn't find it.)

We do have a bug logged for this in GitHub, but it's private since it includes reference to your data. I will ask support to update your SR appropriately.

**Comment by youngbrett** [comment, marked as answer]
Thanks, @mgooding.  

I originally asked about versioning info for iTwin Synchronizer because I didn't know if this info lived somewhere I hadn't found.  Its important because I think that Synchronization and Connectors are a huge part of the value of the iTwin stack (for my area in commercial construction, at least), I still frequently have problems with them, and I perceive versioning info as being helpful for my troubleshooting.  For example, I tried to generate an iModel from Revit files yesterday and it failed with iTwin Synchronization but worked with iTwin Snapshot.  Since this version info does not appear to be publicly accessible, I suggest we mark this as answered and I'll start a new idea thread or otherwise make a proposed feature request. Objections?
